### PR TITLE
Add support for Debian 12

### PR DESCRIPTION
**Pull request by y3n4**
Seemed more robust to base the configuration steps on the installed version of spamassassin instead of the release version for the edge case that someone would run on bullseye but with spamassassin 4.0.0 installed from bullseye-backports, and eventually ended up adding tests for said edge case too

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -19,3 +19,5 @@ jobs:
   integration:
     name: Integration
     uses: systemli/github-ansible-workflow/.github/workflows/ansible-integration-workflow.yaml@v1.2.0
+    with:
+      distros: '[ "debian12", "debian11" ]'
```
Comment (t2d) [inline .github/workflows/main.yml:23]:
```suggestion
```
this is the default. we can just drop this

File: tasks/main.yml [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
   apt:
     pkg: "{{ spamassassin_packages }}"
     state: present
     cache_valid_time: 600
 
+- name: Gather list of installed packages
+  ansible.builtin.package_facts:
+    manager: apt
+
+- name: Gather if >= 4.0.0-1 ist installed
```
Comment (t2d) [inline tasks/main.yml:12]:
```suggestion
- name: Gather if spamassassin >= 4.0.0-1 is installed
```

**Comment by y3n4** [comment]
Fixed the templating in [f8ddf74](https://github.com/systemli/ansible-role-spamassassin/pull/45/commits/f8ddf74c9f7dfd1ad34336ab5ec2c49064083984)

[7555aba](https://github.com/systemli/ansible-role-spamassassin/pull/45/commits/7555abadcbd7113d5c1e15a3e237202cb13a6978) also changes the name of the role variable in a backwards compatible way, to reflect the upstream wording, but I am not sure if this extra logic could have any downsides?

**Comment by t2d** [comment]
I like it a lot!

**Review by t2d — APPROVED** [review]
